clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -198,7 +198,9 @@
 
     /** <code>request.timeout.ms</code> */
     public static final String REQUEST_TIMEOUT_MS_CONFIG = CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG;
-    private static final String REQUEST_TIMEOUT_MS_DOC = CommonClientConfigs.REQUEST_TIMEOUT_MS_DOC;
+    private static final String REQUEST_TIMEOUT_MS_DOC = CommonClientConfigs.REQUEST_TIMEOUT_MS_DOC
+                                                        + " request.timeout.ms should be larger than replica.lag.time.max.ms "
+                                                        + "to reduce message duplication caused by unnecessary producer retry.";
```

Comment:
Hmm, this should be in the producer config, right? Also, could we mention that replica.lag.time.max.ms is a broker side config? Otherwise, people would assume this is another producer side config.